File: id_operateur/app/Models/Document.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Document extends Model
{
    use HasFactory;

    protected $table = 'documents';

    protected $fillable = [
        'personne_id',
        'type_document',
        'numero',
        'fichier',
        'date_expiration',
        'est_valide'
    ];

    protected $casts = [
        'date_expiration' => 'date',
        'est_valide' => 'boolean',
    ];

    // ============================================================
    // RELATIONS
    // ============================================================
    
    public function personne()
    {
        return $this->belongsTo(Personne::class);
    }

    // ============================================================
    // SCOPES
    // ============================================================
    
    public function scopeValide($query)
    {
        return $query->where('est_valide', true);
    }

    public function scopeParType($query, $type)
    {
        return $query->where('type_document', $type);
    }

    public function scopeParPersonne($query, $personneId)
    {
        return $query->where('personne_id', $personneId);
    }

    // ============================================================
    // ACCESSORS
    // ============================================================
    
    public function getTypeDocumentLabelAttribute()
    {
        $labels = [
            'CNI' => 'Carte Nationale d\'Identité',
            'PASSEPORT' => 'Passeport',
            'STATUTS' => 'Statuts',
            'RCCM' => 'Registre de Commerce',
            'PATENTE' => 'Patente',
            'QUITTANCE' => 'Quittance',
            'AVATAR' => 'Photo de profil',
            'AUTRE' => 'Autre',
        ];
        return $labels[$this->type_document] ?? $this->type_document;
    }

    public function getCheminFichierAttribute()
    {
        return storage_path('app/public/' . $this->fichier);
    }

    public function getUrlAttribute()
    {
        return asset('storage/' . $this->fichier);
    }

    // ============================================================
    // MÉTHODES UTILITAIRES
    // ============================================================
    
    public function estValide()
    {
        if (!$this->est_valide) {
            return false;
        }
        if ($this->date_expiration && $this->date_expiration < now()) {
            return false;
        }
        return true;
    }

    public function estExpire()
    {
        return $this->date_expiration && $this->date_expiration < now();
    }
}

File: gs_operateur_mobile/lib/models/document.dart
class Document {
  final int id;
  final int? personneId;
  final String typeDocument;
  final String? numero;
  final String? fichier;
  final bool? estValide;
  final DateTime? createdAt;
  final PersonneRef? personne;

  Document({
    required this.id,
    this.personneId,
    required this.typeDocument,
    this.numero,
    this.fichier,
    this.estValide,
    this.createdAt,
    this.personne,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'] ?? 0,
      personneId: json['personne_id'],
      typeDocument: json['type_document'] ?? json['type'] ?? '',
      numero: json['numero'],
      fichier: json['fichier'],
      estValide: json['est_valide'],
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'])
          : null,
      personne: json['personne'] != null
          ? PersonneRef.fromJson(json['personne'])
          : null,
    );
  }
}

class PersonneRef {
  final int id;
  final String? nomComplet;

  PersonneRef({required this.id, this.nomComplet});

  factory PersonneRef.fromJson(Map<String, dynamic> json) {
    return PersonneRef(
      id: json['id'] ?? 0,
      nomComplet: json['nom_complet'],
    );
  }
}

class DocumentType {
  final String value;
  final String label;

  DocumentType({required this.value, required this.label});

  static List<DocumentType> get all => [
        DocumentType(value: 'CNI', label: 'Carte Nationale d\'Identité'),
        DocumentType(value: 'PASSEPORT', label: 'Passeport'),
        DocumentType(value: 'RCCM', label: 'RCCM'),
        DocumentType(value: 'PATENTE', label: 'Patente'),
        DocumentType(value: 'QUITTANCE', label: 'Quittance'),
        DocumentType(value: 'STATUTS', label: 'Statuts'),
        DocumentType(value: 'AVATAR', label: 'Photo'),
        DocumentType(value: 'AUTRE', label: 'Autre'),
      ];
}
